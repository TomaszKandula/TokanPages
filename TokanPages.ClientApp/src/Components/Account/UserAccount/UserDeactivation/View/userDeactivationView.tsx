import * as React from "react";
import { Button, CircularProgress, Grid, Typography } from "@material-ui/core";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Container from "@material-ui/core/Container";
import Skeleton from "@material-ui/lab/Skeleton";
import { ViewProperties } from "../../../../../Shared/Abstractions";
import { SectionAccountDeactivation } from "../../../../../Api/Models";
import { CustomDivider, RenderParagraphs } from "../../../../../Shared/Components";
import { UserDeactivationProps } from "../userDeactivation";

interface UserDeactivationViewProps extends ViewProperties, UserDeactivationProps {
    buttonHandler: () => void;
    progress: boolean;
    section: SectionAccountDeactivation;
}

const DeactivationButton = (props: UserDeactivationViewProps): React.ReactElement => {
    return (
        <Button
            fullWidth
            type="submit"
            variant="contained"
            onClick={props.buttonHandler}
            disabled={props.progress}
            className="button-delete"
        >
            {!props.progress ? props.section?.deactivateButtonText : <CircularProgress size={20} />}
        </Button>
    );
};

export const UserDeactivationView = (props: UserDeactivationViewProps): React.ReactElement => {
    return (
        <section className={`section ${props.background ?? ""}`}>
            <Container className="container-wide">
                <div className="pb-40">
                    <Card elevation={0} className="card">
                        <CardContent className="card-content">
                            <Typography component="span" className="caption black">
                                {props.isLoading ? <Skeleton variant="text" /> : props.section?.caption}
                            </Typography>
                            <CustomDivider mt={15} mb={8} />
                            <div className="pt-8 pb-8">
                                <Grid container spacing={2}>
                                    <Grid item>
                                        {props.isLoading ? (
                                            <Skeleton variant="text" width="200px" />
                                        ) : (
                                            <RenderParagraphs text={props.section?.warningText} className="label user-account-text-item" />
                                        )}
                                    </Grid>
                                </Grid>
                                <CustomDivider mt={15} mb={15} />
                                <Grid className="button-container-update">
                                    <div className="mt-15 mb-15">
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="150px" height="40px" />
                                        ) : (
                                            <DeactivationButton {...props} />
                                        )}
                                    </div>
                                </Grid>
                            </div>
                        </CardContent>
                    </Card>
                </div>
            </Container>
        </section>
    );
};
