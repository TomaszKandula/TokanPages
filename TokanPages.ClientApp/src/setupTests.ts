// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
