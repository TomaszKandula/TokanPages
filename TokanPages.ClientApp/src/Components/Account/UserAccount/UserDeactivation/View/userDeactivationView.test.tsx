import "../../../../../setupTests";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { UserDeactivationView } from "./userDeactivationView";

describe("test account group component: userDeactivationView", () => {
    it("should render correctly '<UserDeactivationView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <UserDeactivationView
                    isLoading={false}
                    isMobile={false}
                    progress={false}
                    buttonHandler={jest.fn()}
                    section={{
                        caption: "Account Deactivation",
                        warningText: ["Warning!", "You can deactivate the account..."],
                        deactivateButtonText: "Dactivate",
                        deactivatePromptText: "Are you sure?"
                    }}
                />
            </BrowserRouter>
        );

        expect(html).toMatchSnapshot();
    });
});
