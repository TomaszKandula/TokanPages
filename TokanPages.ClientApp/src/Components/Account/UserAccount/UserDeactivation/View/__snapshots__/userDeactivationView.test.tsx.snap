// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test account group component: userDeactivationView should render correctly '<UserDeactivationView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="section "
      >
        <div
          class="MuiContainer-root container-wide MuiContainer-maxWidthLg"
        >
          <div
            class="pb-40"
          >
            <div
              class="MuiPaper-root MuiCard-root card MuiPaper-elevation0 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root card-content"
              >
                <span
                  class="MuiTypography-root caption black MuiTypography-body1"
                >
                  Account Deactivation
                </span>
                <div
                  class="mt-15 mb-8"
                >
                  <hr
                    class="MuiDivider-root divider"
                  />
                </div>
                <div
                  class="pt-8 pb-8"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <span
                        class="MuiTypography-root label MuiTypography-body1"
                      >
                        &lt;p&gt;You can deactivate the account...&lt;/p&gt;
                      </span>
                    </div>
                  </div>
                  <div
                    class="mt-15 mb-15"
                  >
                    <hr
                      class="MuiDivider-root divider"
                    />
                  </div>
                  <div
                    class="MuiGrid-root button-container-update"
                  >
                    <div
                      class="mt-15 mb-15"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained button-delete MuiButton-fullWidth"
                        tabindex="0"
                        type="submit"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          Dactivate
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="section "
    >
      <div
        class="MuiContainer-root container-wide MuiContainer-maxWidthLg"
      >
        <div
          class="pb-40"
        >
          <div
            class="MuiPaper-root MuiCard-root card MuiPaper-elevation0 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root card-content"
            >
              <span
                class="MuiTypography-root caption black MuiTypography-body1"
              >
                Account Deactivation
              </span>
              <div
                class="mt-15 mb-8"
              >
                <hr
                  class="MuiDivider-root divider"
                />
              </div>
              <div
                class="pt-8 pb-8"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <span
                      class="MuiTypography-root label MuiTypography-body1"
                    >
                      &lt;p&gt;You can deactivate the account...&lt;/p&gt;
                    </span>
                  </div>
                </div>
                <div
                  class="mt-15 mb-15"
                >
                  <hr
                    class="MuiDivider-root divider"
                  />
                </div>
                <div
                  class="MuiGrid-root button-container-update"
                >
                  <div
                    class="mt-15 mb-15"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained button-delete MuiButton-fullWidth"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Dactivate
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
